pkg/utils/imagevector/types.go

```diff
@@ -25,6 +34,7 @@ type ImageSource struct {
 	Name           string  `json:"name" yaml:"name"`
 	RuntimeVersion *string `json:"runtimeVersion,omitempty" yaml:"runtimeVersion,omitempty"`
 	TargetVersion  *string `json:"targetVersion,omitempty" yaml:"targetVersion,omitempty"`
+	Architecture   *string `json:"architecture,omitempty" yaml:"architecture,omitempty"`
```

Comment:
Can you elaborate why it is required to make the imagevector pkg architecture aware? Wasn't the idea to have multi-arch images in `charts/images.yaml`? Or did I get it wrong?
Doesn't this change also result in changes in `charts/images.yaml`? As part of this PR I don't see that we specify the new architecture field for any of the images.

Reply:
> Can you elaborate why it is required to make the imagevector pkg architecture aware? Wasn't the idea to have multi-arch images in charts/images.yaml? Or did I get it wrong?

Our goal is to have as many possible multi arch images but there is a chance we can't achieve it. As discussed with @rfranzke we need to add a new field.

```
I think it’s questionable whether we can achieve having only multi-arch images since there are some
images that we rely on which are out of our control (if community does not help in timely manner then what?). 
But even if we make it, what if we introduce a new image tomorrow where only separate images are available and community does not want to build multi-arch image for whatever reasons? In this case, we have an issue with ARM shoots
```
> Doesn't this change also result in changes in charts/images.yaml? As part of this PR I don't see that we specify the new architecture field for any of the images.

Yes, the changes in `charts/images.yaml` are required. If an image doesn't have arch filed it is by default considered as amd64 based image. I will update the `image.yaml` once the PR is ready for review based on what are images available at that time.